### PR TITLE
feat: Add three dot menu to the post

### PR DESCRIPTION
**Pull request by arifjahan88**
Add a three-dot menu button to the post. #15

### CODE REVIEW

**Comment by shahmedme** [comment]
We can add some functionalities on dropdown too. These can be added,
- Edit post
- Delete post
- Report post

More suggestions are appreciated.

**Comment by arifjahan88** [comment]
@shahmedme Added dropdown menu. #15
image

**Comment by arifjahan88** [comment]
@shahmedme vai edit and deleteModal added. olease check #15 

image
image

**Comment by shahmedme** [comment]
Thanks for the PR. It's a nice addition. Here're some feedbacks:

- Only post owner can use only Edit and Delete action.
- Only other users can use Report post action.
- Need to add API connection too for Edit and Delete post.
- Can we make modal size (width, height) same as create post modal?
- Let's have a gray background while hovering over three dot button.
- Also we can place delete post confirmation modal vertically centered.

File: web/src/pages/feed/views/Feed/PostSingle.tsx [hunk 1]
```diff
@@ -37,6 +47,19 @@ export default function PostSingle(props: PostT) {
...
+  const handlethreeDotsClick = (key: string) => {
+    if (!user) {
+      dispatch(requireAuth());
+      return;
+    }
+
+    if (key === 'edit_post') {
+      dispatch(handleEditPostModal({ open: true, id: id }));
+    } else if (key === 'delete_post') {
+      dispatch(handleDeletePostModal({ open: true, id: id }));
```
Comment (shahmedme) [inline web/src/pages/feed/views/Feed/PostSingle.tsx:59]: For delete we can use Modal.confirm() from antd

File: web/src/pages/feed/views/Feed/PostSingle.tsx [hunk 2]
```diff
@@ -59,6 +82,34 @@ export default function PostSingle(props: PostT) {
...
+                  label: <span>Report Post</span>,
+                  key: 'report_post',
+                  icon: <MdOutlineReport size={15} />,
+                  danger: true,
+                },
+              ],
+              onClick: (event) => handlethreeDotsClick(event.key),
+            }}
+          >
+            <BsThreeDotsVertical className="text-xs text-neutral-500 cursor-pointer" />
```
Comment (shahmedme) [inline web/src/pages/feed/views/Feed/PostSingle.tsx:111]: Can we put gray background while hover?

**Review by shahmedme — CHANGES_REQUESTED** [review]
Thanks for the PR. It's a nice addition. Here're some feedbacks:

- Only post owner can use only Edit and Delete action.
- Only other users can use Report post action.
- Need to add API connection too for Edit and Delete post.
- Can we make modal size (width, height) same as create post modal?
- Let's have a gray background while hovering over three dot button.
- Also we can place delete post confirmation modal vertically centered.

File: web/src/pages/feed/views/Feed/PostSingle.tsx (hunk 2 above)
Comment (arifjahan88) [reply web/src/pages/feed/views/Feed/PostSingle.tsx:111]: added

File: web/src/pages/feed/views/Feed/PostSingle.tsx [hunk 3]
```diff
@@ -37,6 +47,19 @@ export default function PostSingle(props: PostT) {
     share_count,
   } = props;
 
+  const handlethreeDotsClick = (key: string) => {
```
[the remark this answers is not in the document]
Comment (arifjahan88) [reply web/src/pages/feed/views/Feed/PostSingle.tsx:50]: done

**Pull request closed without merging**